route.go

```diff
@@ -28,3 +34,178 @@ func (r *Route) String() string {
 func (r *Route) GetFullPath() string {
 	return r.fullPath
 }
+
+func (route *Route) getCompletePath() {
+	var parts []string
+	currentRouter := route.Router
+	for currentRouter != nil {
+		parts = append([]string{currentRouter.Prefix}, parts...)
+		currentRouter = currentRouter.parent
+	}
+
+	parts = append(parts, route.Path)
+	route.fullPath = strings.Join(parts, "")
+}
+
+func (route *Route) createRegexMatch() {
+	escapedPath := strings.ReplaceAll(route.fullPath, "/", "\\/")
+	regexPattern := regexp.MustCompile(`\{[^}]+\}`).ReplaceAllString(escapedPath, "([^/]+)")
+	route.regexp = regexp.MustCompile("^" + regexPattern + "$")
+}
+
+func (route *Route) handleInputSchema() error { // should this return an error or should it panic?
```

Comment:
oh my golly gosh why couldn't we keep that in field.go. field.go is meant to handle input schemas. i see the overlapping purpose but field.go is a master at this exact thing.